test/lib/jdk/test/lib/hotspot/ir_framework/TestFramework.java

```diff
@@ -0,0 +1,1226 @@
+
+    /**
+     * Verifies that {@code m} is compiled with C1.
+     *
+     * @param m the method to be verified.
+     * @throws TestRunException if {@code m} is not compiled with C1.
+     */
+    public static void assertCompiledByC1(Method m) {
+        TestFrameworkExecution.assertCompiledByC1(m);
+    }
+
+    /**
+     * Verifies that {@code m} is compiled with C2.
+     *
+     * @param m the method to be checked.
+     * @throws TestRunException if {@code m} is not compiled with C2.
+     */
+    public static void assertCompiledByC2(Method m) {
+        TestFrameworkExecution.assertCompiledByC2(m);
+    }
+
+    /**
+     * Verifies that {@code m} is compiled at the specified {@code compLevel}.
+     *
+     * @param m the method to be checked.
+     * @param compLevel the compilation level.
+     * @throws TestRunException if {@code m} is not compiled at {@code compLevel}.
+     */
+    public static void assertCompiledAtLevel(Method m, CompLevel compLevel) {
+        TestFrameworkExecution.assertCompiledAtLevel(m, compLevel);
+    }
+
+    /**
+     * Verifies that {@code m} was deoptimized after being C1 compiled.
+     *
+     * @param m the method to be checked.
+     * @throws TestRunException if {@code m} is was not deoptimized after being C1 compiled.
+     */
+    public static void assertDeoptimizedByC1(Method m) {
+        TestFrameworkExecution.assertDeoptimizedByC1(m);
+    }
+
+    /**
+     * Verifies that {@code m} was deoptimized after being C2 compiled.
+     *
+     * @param m the method to be checked.
+     * @throws TestRunException if {@code m} is was not deoptimized after being C2 compiled.
+     */
+    public static void assertDeoptimizedByC2(Method m) {
+        TestFrameworkExecution.assertDeoptimizedByC2(m);
+    }
+
+    /*
+     * End of public interface methods
+     */
+
+    /**
+     * Used to move Whitebox class to the right folder in the JTreg test
+     */
+    private void installWhiteBox() {
+        try {
+            ClassFileInstaller.main(WhiteBox.class.getName());
+        } catch (Exception e) {
+            throw new Error("failed to install whitebox classes", e);
+        }
+    }
+
+    /**
+     * Disable IR verification completely in certain cases.
+     */
+    private void maybeDisableIRVerificationCompletely() {
+        if (VERIFY_IR) {
+            VERIFY_IR = hasIRAnnotations();
+            if (!VERIFY_IR) {
+                System.out.println("IR verification disabled due to test " + testClass + " not specifying any @IR annotations");
+                return;
+            }
+
+            VERIFY_IR = Platform.isDebugBuild() && !Platform.isInt() && !Platform.isComp();
+            if (!VERIFY_IR) {
+                System.out.println("IR verification disabled due to not running a debug build (required for PrintIdeal" +
+                                   "and PrintOptoAssembly), running with -Xint, or -Xcomp (use warm-up of 0 instead)");
+                return;
+            }
```

Comment:
I'd reorder them as "platform" checks are faster than `hasIRAnnotations` check and, I'd guess, will be a more common culprit to disable IR verification.